pkgs/applications/version-management/sublime-merge/common.nix

```diff
@@ -0,0 +1,107 @@
+{ buildVersion, sha256, dev ? false }:
+
+{ fetchurl, stdenv, xorg, glib, gtk3, cairo, pango, libredirect, makeWrapper, wrapGAppsHook
+, pkexecPath ? "/run/wrappers/bin/pkexec", gksuSupport ? false, gksu
+, writeScript, common-updater-scripts, curl, gnugrep
+}:
+
+assert gksuSupport -> gksu != null;
+
+let
+  libPath = stdenv.lib.makeLibraryPath [xorg.libX11 glib gtk3 cairo pango];
+  redirects = [ "/usr/bin/pkexec=${pkexecPath}" ]
+    ++ stdenv.lib.optional gksuSupport "/usr/bin/gksudo=${gksu}/bin/gksudo";
+in let
+  # package with just the binaries
+  sublime_merge = stdenv.mkDerivation {
+    name = "sublime-merge-${buildVersion}-bin";
+    version = buildVersion;
+
+    src = fetchurl {
+      name = "sublime-merge-${buildVersion}.tar.xz";
+      url = "https://download.sublimetext.com/sublime_merge_build_${buildVersion}_x64.tar.xz";
+      inherit sha256;
+    };
+
+    dontStrip = true;
+    dontPatchELF = true;
+    buildInputs = [ glib gtk3 ]; # for GSETTINGS_SCHEMAS_PATH
+    nativeBuildInputs = [ makeWrapper wrapGAppsHook ];
+
+    buildPhase = ''
+      runHook preBuild
+
+      for binary in sublime_merge crash_reporter git-credential-sublime ssh-askpass-sublime; do
+        patchelf \
+          --interpreter "$(cat $NIX_CC/nix-support/dynamic-linker)" \
+          --set-rpath ${libPath}:${stdenv.cc.cc.lib}/lib64 \
+          $binary
+      done
+
+      # Rewrite pkexec|gksudo argument. Note that we can't delete bytes in binary.
+      sed -i -e 's,/bin/cp\x00,cp\x00\x00\x00\x00\x00\x00,g' sublime_merge
+
+      runHook postBuild
+    '';
+
+    installPhase = ''
+      runHook preInstall
+
+      # Correct sublime_merge.desktop to exec 'sublime_merge' instead of /opt/sublime_merge
+      substituteInPlace "sublime_merge.desktop" --replace "/opt/sublime_merge/sublime_merge" "$out/sublime_merge"
+
+      mkdir -p $out
+      cp -r * $out/
+
+      runHook postInstall
+    '';
+
+    dontWrapGApps = true; # non-standard location, need to wrap the executables manually
+
+    postFixup = ''
+      wrapProgram $out/sublime_merge \
+        --set LD_PRELOAD "${libredirect}/lib/libredirect.so" \
+        --set NIX_REDIRECTS ${builtins.concatStringsSep ":" redirects} \
+        "''${gappsWrapperArgs[@]}"
+    '';
+  };
+in stdenv.mkDerivation {
+  name = "sublime-merge-${buildVersion}";
```

Comment:
```suggestion
  pname = "sublime-merge";
```

Reply:
Hi @tadeokondrak, thanks for the feedback. I was having trouble finding documentation on exactly when to use `pname` vs `name`. Am I correct in understanding that you should use `pname` when you have a separate `version` attribute and `name` when the version is part of the name?

Reply:
Yeah, we need documentation on this really soon.

Simply put, the `name` attribute is the derivation name. And when attributes `pname` and `version` are present they get concatenated to `name = "${pname}-${version}"`

> https://github.com/NixOS/nixpkgs/blob/c40f2114992f7387013a6113ebfd99ae63ec4f91/pkgs/stdenv/generic/make-derivation.nix#L181

This exists because you frequently want to reuse the value of `pname` and `version` throughout the expression.